PrebidMobile/PrebidMobileRendering/Utilities/PBMUserAgentService.m

```diff
@@ -68,6 +70,13 @@ - (instancetype)init {
 #pragma mark - Public Methods
 
 - (nonnull NSString *)getFullUserAgent {
+    if (self.userAgent == nil) {
+        self.uaSemaphore = dispatch_semaphore_create(0);
```

Comment:
If two ad units call this method at the same time before ua is obtained, the semaphore will be repeated, with unpredictable consequences

Reply:
I'm not sure about using semaphore here. It is a tool dedicated to multithreading access to resources. But we need just to lock the process for any resource consumer. So, for me, using NSLock will be much more convenient here. It will let us avoid using the while loop for waiting for the resource. But we still need to be sure of proper unlocking.

Reply:
NSLock doesn't lock anything itself. But yes, the requirement that lock() and unlock() should be called on the same thread makes it hardly usable in this situation. 

The current implementation looks good. However, this [loop](dispatch_semaphore_wait) looks potentially endless. In the case when the semaphore will be actually incremented before the start of the loop, the wait will always be finished with a timeout. Does it make sense to add an extra check for userAgent inside the loop or limit the number of possible loops?